pkg/controller/servicebindingrequest/reconciler.go

```diff
@@ -194,6 +200,23 @@ func (r *Reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 		return sb.Unbind()
 	}
 
+	var emptyApplication v1alpha1.ApplicationSelector
+	if sbr.Spec.ApplicationSelector == emptyApplication ||
+		(sbr.Spec.ApplicationSelector.ResourceRef == "" &&
+			sbr.Spec.ApplicationSelector.LabelSelector.MatchLabels == nil) {
+		conditionsv1.SetStatusCondition(&sbr.Status.Conditions, conditionsv1.Condition{
+			Type:    InjectionReady,
+			Status:  corev1.ConditionFalse,
+			Reason:  EmptyApplicationSelectorReason,
+			Message: "The spec.applicationSelector field is empty.",
+		})
+		_, updateErr := updateServiceBindingRequestStatus(r.dynClient, sbr)
+		if updateErr == nil {
+			return Done()
+		}
```

Comment:
What do you think about creating a function combining `conditionsv1.SetStatusConditions` and `updateServiceBindingRequestStatus`? I believe it'd be useful right away to deal with conditions everywhere in the operator.

Reply:
@isutton 

You mean passing the condition to `updateServiceBindingReuqest` like this:

```
		condition := conditionsv1.Condition{
			Type:    CollectionReady,
			Status:  corev1.ConditionFalse,
			Reason:  EmptyServiceSelectorsReason,
			Message: "The spec.backingServiceSelectors field is empty.",
		}
		_, updateErr := updateServiceBindingRequestStatus(r.dynClient, sbr, condition)
```

Reply:
I would change it to a variadic function accepting multiple conditions.

Reply:
Resolved 
d32bed7